src/main/java/com/adyen/model/terminal/AmountsResp.java

```diff
@@ -0,0 +1,324 @@
+ * 
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+
+package com.adyen.model.terminal;
+
+import java.util.Objects;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.HashMap;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.annotation.JsonValue;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import java.math.BigDecimal;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+
+/**
+ * AmountsResp
+ */
+@JsonPropertyOrder({
+  AmountsResp.JSON_PROPERTY_CURRENCY,
+  AmountsResp.JSON_PROPERTY_AUTHORIZED_AMOUNT,
+  AmountsResp.JSON_PROPERTY_TOTAL_REBATES_AMOUNT,
+  AmountsResp.JSON_PROPERTY_TOTAL_FEES_AMOUNT,
+  AmountsResp.JSON_PROPERTY_CASH_BACK_AMOUNT,
+  AmountsResp.JSON_PROPERTY_TIP_AMOUNT
+})
+
+public class AmountsResp {
+  public static final String JSON_PROPERTY_CURRENCY = "Currency";
+  private String currency;
+
+  public static final String JSON_PROPERTY_AUTHORIZED_AMOUNT = "AuthorizedAmount";
+  private BigDecimal authorizedAmount;
+
+  public static final String JSON_PROPERTY_TOTAL_REBATES_AMOUNT = "TotalRebatesAmount";
+  private BigDecimal totalRebatesAmount;
+
+  public static final String JSON_PROPERTY_TOTAL_FEES_AMOUNT = "TotalFeesAmount";
+  private BigDecimal totalFeesAmount;
+
+  public static final String JSON_PROPERTY_CASH_BACK_AMOUNT = "CashBackAmount";
+  private BigDecimal cashBackAmount;
+
+  public static final String JSON_PROPERTY_TIP_AMOUNT = "TipAmount";
+  private BigDecimal tipAmount;
+
+  public AmountsResp() { 
+  }
+
+  public AmountsResp currency(String currency) {
+    this.currency = currency;
+    return this;
+  }
+
+   /**
+   * Get currency
+   * @return currency
+  **/
+  @ApiModelProperty(value = "")
+  @JsonProperty(JSON_PROPERTY_CURRENCY)
+  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)
+
+  public String getCurrency() {
+    return currency;
+  }
+
+
+ /**
+  * currency
+  *
+  * @param currency
+  */ 
+  @JsonProperty(JSON_PROPERTY_CURRENCY)
+  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)
+  public void setCurrency(String currency) {
+    this.currency = currency;
+  }
+
+
+  public AmountsResp authorizedAmount(BigDecimal authorizedAmount) {
+    this.authorizedAmount = authorizedAmount;
+    return this;
+  }
+
+   /**
+   * Get authorizedAmount
+   * minimum: 0.0
+   * maximum: 99999999.999999
+   * @return authorizedAmount
+  **/
+  @ApiModelProperty(required = true, value = "")
+  @JsonProperty(JSON_PROPERTY_AUTHORIZED_AMOUNT)
+  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)
+
+  public BigDecimal getAuthorizedAmount() {
```

Comment:
This type seems fishy. isn't amount always an integer or long?

Reply:
Actually we already used this type in the old nexo models, let me double check tho